scapy/contrib/mqtt.py

```diff
@@ -219,15 +219,33 @@ class MQTTPubcomp(Packet):
         ShortField("msgid", None),
     ]
 
+class MQTTTopic(Packet):
+    name = "MQTT topic"
+    fields_desc = [
+        FieldLenField("length", None, length_of="topic"),
+        StrLenField("topic", "", length_from=lambda pkt:pkt.length)
+    ]
+
+    def guess_payload_class(self, payload):
+        return conf.padding_layer
+
+class MQTTTopicQOS(MQTTTopic):
+    fields_desc = MQTTTopic.fields_desc + [ByteEnumField("QOS", 0, QOS_LEVEL)]
+
+def cb_topic_qos(pkt, lst, cur, remain):
+    """
+    Decode the remaining bytes as a MQTT topic
+    """
+    if len(remain) > 3:
+        return MQTTTopicQOS
+    else:
+        return conf.raw_layer
 
 class MQTTSubscribe(Packet):
     name = "MQTT subscribe"
     fields_desc = [
         ShortField("msgid", None),
-        FieldLenField("length", None, length_of="topic"),
-        StrLenField("topic", "",
-                    length_from=lambda pkt: pkt.length),
-        ByteEnumField("QOS", 0, QOS_LEVEL),
+        PacketListField("topics", [], next_cls_cb=cb_topic_qos)
```

Comment:
I don't think you need the `cb_topic_qos` wrapper; this should work:
```suggestion
        PacketListField("topics", [], next_cls_cb=MQTTTopicQOS)
```

Reply:
@p-l- Hello! I'm sorry but your suggestion seems to be not working for me. The cb_topic_qos is necessary in order to properly identify the remaining bytes (the topic length, the topic name, and the QoS level). Without that function cb_topic_qos , the fields can not be properly identified.

Reply:
Sorry, I have answered too fast. This should work (without the need for a function), or do I miss something again?
```suggestion
        PacketListField("topics", [], cls=MQTTTopicQOS)
```

Reply:
@p-l- Thank you so much! It worked! I've also removed the function cb_topic, which was also unnecessary for Unsubscribe packets. Is that ok? I've also added unit tests for both Unsubscribe and Unsuback packets in the latest pull request.

Reply:
@p-l- Please don't use the key argument `cls=` because it will be restricted when we will start using Type hints (https://github.com/secdev/scapy/pull/2671) because of a stupid restriction in `typing.Generic`. Use it as positional argument (you can use any other key arguments freely, just not `cls`)

Reply:
@p-l- Considering the suggestion made by @gpotter2 , do you have any suggestions on what we can do instead of using the key argument cls= ? Will the functions cb_topic and cb_topic_qos be necessary?
Thank you for your time!

Reply:
Just use `PacketListField("topics", [], MQTTTopicQOS)`

Reply:
@gpotter2 Thank you very much for your suggestion! I removed the key argument cls= , and updated the PR.

Reply:
@gpotter2 since `cls` is defined as a keyword argument, my opinion is that we **should** try to always use it as a keyword argument (even though that's not a requirement for Python).

So we probably want to rename it (`cls` is usually the `self` for classmethods, so that's not a good choice anyway) in another PR. In the meantime, I would prefer that we use `cls=`, so that, when we rename the argument name, we can find this one with al the other ones and change it.

Reply:
Hello. Considering the suggestion by @gpotter2 , is it ok if I update the PR with `cls=` ?
Thank you for your time!

Reply:
You can.

Reply:
Thank you! I updated the PR.
Please let me know if anything else is necessary.